File: leetcode/prefix-sum/py/1732.find-the-highest-altitude.py
#
# @lc app=leetcode id=1732 lang=python3
#
# [1732] Find the Highest Altitude
#

# @lc code=start
class Solution:
	def largestAltitude(self, gain: List[int]) -> int:
		maxHeight, height = 0, 0
		for num in gain:
			height += num
			maxHeight = max(height, maxHeight)
		return maxHeight
# @lc code=end

File: leetcode/1732.find-the-highest-altitude.cpp
/*
 * @lc app=leetcode id=1732 lang=cpp
 *
 * [1732] Find the Highest Altitude
 */

// @lc code=start
class Solution {
public:
	int largestAltitude(vector<int>& gain) {
		int	max = 0;
		int height = 0;

		for (auto num : gain) {
			height += num;
			max = max > height ? max : height;
		}
		return max;
	}
};
// @lc code=end

